### PR TITLE
Unify xen modules, so there is only one set (in shell/components/xen)

### PR DESCRIPTION
**Pull request by sjmiles**
Also removing the old files in (browser/lib/).

### CODE REVIEW

File: shell/components/x-list.js [hunk 1]
```diff
@@ -8,11 +8,16 @@
...
  */
 
-import XTemplate from './xen-template.js';
-import XElement from './xen-element.js';
-import XState from './xen-state.js';
+import Xen from './xen/xen.js';
 
-class XList extends XState(XElement) {
+let HTMLElement;
+if (typeof window == 'undefined') {
```
Comment (shaper) [inline shell/components/x-list.js]: nit: I usually see `===` for these?

File: shell/components/xen/xen-state.js [hunk 2]
```diff
@@ -51,8 +51,8 @@ let StaterMixin = Base => class extends Base {
   }
   _async(fn) {
     // TODO(sjmiles): SystemJS throws unless `Promise` is `window.Promise`
-    //return Promise.resolve().then(fn.bind(this));
-    return setTimeout(fn.bind(this), 10);
+    return Promise.resolve().then(fn.bind(this));
```
Comment (shaper) [inline shell/components/xen/xen-state.js]: I mentioned this to @mmandlis today but I didn't have full context -- is there an issue that would be useful to file to track whatever ramification this has for runtime logic?

File: shell/components/x-list.js (hunk 1 above)
Comment (sjmiles) [reply shell/components/x-list.js]: When testing against a literal string, there is no difference between `==` and `===`, but I can change it if it's bothersome.

File: shell/components/xen/xen-state.js (hunk 2 above)
Comment (sjmiles) [reply shell/components/xen/xen-state.js]: Perhaps, I wanted to see the failure one more time before posting an issue. Also, I'm generally concerned about spending time on things that aren't going to matter in the fullness of time (iow, it's likely there are bigger fish to fry).

**Pull request merged**